client/my-sites/promote-post-i2/components/campaigns-table/index.tsx

```diff
@@ -24,9 +46,27 @@ export default function CampaignsTable( props: Props ) {
 					</tr>
 				</thead>
 				<tbody>
-					{ campaigns.map( ( campaign ) => {
-						return <CampaignItem key={ `item-${ campaign.campaign_id }` } campaign={ campaign } />;
-					} ) }
+					{ isLoading && ! isFetchingPageResults ? (
+						<>
+							<CampaignItemLoading />
```

Comment:
Not in this PR, but we need to iterate on this idea. It could be nice to have the same number of posts from the last call (with a maximum) as loading when performing another filter.

Right now, if you have less than 4 posts, the experience is a little strange, you see that the page gains a vertical scroll and then comes back only with a single post. 

But its better to start a discussion with Bart to see how he wants to do it.

Reply:
Sounds nice, just to give some ideas:

Since it's a loader, doing calls to any specific service would fail the purpose of adding this loader, but we might be able to get an approximate number of posts from the calypso state, and if we don't have it, just default to a number of rows.

Anyways, faking the data with skeletons usually creates this unusual situation.